utilities/transactions/pessimistic_transaction_db.cc

```diff
@@ -600,7 +601,39 @@ void WritePreparedTxnDB::AddCommitted(uint64_t prepare_seq,
       {
         WriteLock wl(&snapshots_mutex_);
         InstrumentedMutex(db_impl_->mutex());
-        snapshots_ = db_impl_->snapshots().GetAll();
+        // We only care about snapshots lower then max
+        auto all_snapshots =
```

Comment:
`all_snapshot` is not used if `curr_seq == snapshots_version_`. Move it into the if-block.
Also I think you only need to hold db_impl_->mutex() for snapshots()->GetAll(), but not the rest of logic?